### ISSUE TITLE
Port 196 GPS to Canvas

### ISSUE THREAD

**Issue by onox** [Juanvvc/c172p-detailed#662]
A second 196 has been added (black instead of white) with an empty screen. Go to [`Models/c172p.xml`](https://github.com/Juanvvc/c172p-detailed/blob/master/Models/c172p.xml#L103-L151) and unhide the Canvas model. After that add an actual Canvas implementation.

See #392 for the discussion.

**Comment by gilbertohasnofb** [comment]
@stuartbuchanan wrote me in an email:

> I don't know if I raised this before, but the Garmin 196 model is
> very expensive and slows down start times a lot - it adds around 10
> seconds to the start time on my machine.  Is it still useful, or
> should we perhaps retire it or replace it with the Canvas version?

@wlbragg @legoboyvdlp @dany93 @onox @Juanvvc @wkitty42 Does anyone know about the state of the canvas version of the Garmin 196? I think last time I checked it was not functional or was less functional than the non-canvas version, which is what we used.

**Comment by wkitty42** [comment]
On 4/1/20 5:23 PM, Gilberto Agostinho wrote:
> @wlbragg @legoboyvdlp @dany93 @onox @Juanvvc @wkitty42 
> Does anyone know about the state of the canvas version of the Garmin 196? I
> think last time I checked it was not functional or was less functional than
> the non-canvas version, which is what we used.

i have no clue on the state of the canvas version... i'm not sure if there is 
any slow down with the current one or not... i've only gotten it out and powered 
it on for a play a couple of times... since i generally run a debug build of FG, 
everything is slower than normal... sorry...

**Comment by stuartbuchanan** [comment]
Hi Folks,

What would the impact be of removing the 196 completely?

Do we think anyone actually uses it in the simulator?  It's a very old
piece of technology, and I suspect that people just use a moving map
instead if they want a GPS.

-Stuart

**Comment by gilbertohasnofb** [comment]
I think there are some people using it, there are a couple of hits when I search the forum for "c172p gps" or "172 gps" or similar things. But if the impact in performance is so big, that might be something to be considered. Stuart, is the impact only on load times or also in the performance in the sim without engabling the GPS?

**Comment by wlbragg** [comment]
>Stuart, is the impact only on load times or also in the performance in the sim without enabling the GPS?

Yes, I would want to know the answer to that question as well.

It's been so long since we looked at using the Canvas GPS I don't remember why it wasn't used initially. Seems to me it wasn't complete or maybe waiting on some canvas helpers that were not yet created?

Is there any working "canvas" 196 to use as a guide for hooking it up?

**Comment by stuartbuchanan** [comment]
Hi Folks,

On my system I perhaps see a 1fps drop as well as the load time impact.
But that's with a fairly big graphics card (GTX 1660).  It would be worth
testing with other systems as well I think.

I've no idea if there's an alternative 196 implementation.

Unfortunately there's no simple way to make it completely optional without
creating a new -set.xml file and a new model file. :(

-Stuart

**Comment by wlbragg** [comment]
Right, I had to do that with the FG1000 in the J3Cub as you can't remove the overhead without completely removing include commands.
I suppose we could make one more .set that includes the GPS as is and remove it from all the other variants. Basically make the GPS equipped aircraft a variant?
What is the overhead, nasal? What about on-demand nasal loading, would that be possible?

**Comment by gilbertohasnofb** [comment]
@wlbragg I think using multiple set files is a good choice, from what I could see in the forum a few people (though not many) use the GPS, so it would be a pity for those users to lose functionality. But the difference in performance might be really welcomed by the majority of users, so I think this is worth it.
